### PR TITLE
380 traceback partition

### PR DESCRIPTION
**Pull request by jimmykarily**
Fixes https://github.com/kairos-io/kairos/issues/380

### CODE REVIEW

**Comment by jimmykarily** [comment]
Though this should fix the original problem, I'm keeping this PR in draft while I'm trying to setup a test suite on this project. I may split it on another PR if it proves to be too much work.

**Comment by mudler** [comment]
with this we don't store anywhere the partition mapping right?

File: cmd/discovery/main.go [hunk 1]
```diff
@@ -103,13 +106,12 @@ func start() error {
...
 		// TODO: This should be 1 call, send both name and label to controller
-		pass, err := waitPass(b.Label, 30)
-		if err != nil || pass == "" {
-			pass, err = waitPass(b.Name, 30)
-			if err != nil {
-				return pluggable.EventResponse{
-					Error: fmt.Sprintf("failed getting pass: %s", err.Error()),
-				}
+		// TODO: Send over all partition data and let the server decide which one
+		// to use to find the partition.
```
Comment (jimmykarily) [inline cmd/discovery/main.go:110]: To be removed

**Comment by mudler** [comment]
> > with this we don't store anywhere the partition mapping right?
> 
> Right. Looking at the code, I think all 3 headers will be available all the time, so we take the first that matches. But I need to verify this in a final booted iso. Up to now I've only tested it by running the challenger binary locally.

It won't be always the case - only the label is the 'minimum' that is set from the user - all the rest is generated on the node after install

**Comment by jimmykarily** [comment]
> > > with this we don't store anywhere the partition mapping right?
> > 
> > 
> > Right. Looking at the code, I think all 3 headers will be available all the time, so we take the first that matches. But I need to verify this in a final booted iso. Up to now I've only tested it by running the challenger binary locally.
> 
> this is fine for local iteration but It won't be always the case - only the label is the 'minimum' that is set from the user - all the rest is generated on the node after install

True. Still the code would still work then. If there is only the label, it will use the label.

File: config/crd/bases/keyserver.kairos.io_sealedvolumes.yaml [hunk 2]
```diff
@@ -37,15 +37,32 @@ spec:
...
+                      properties:
+                        name:
+                          type: string
+                        path:
+                          type: string
+                      type: object
+                    uuid:
+                      type: string
+                  required:
+                  - deviceName
```
Comment (mudler) [inline config/crd/bases/keyserver.kairos.io_sealedvolumes.yaml:60]: this doesn't look right, are all of them required? miss `omitempty` in the spec struct probably

**Comment by mudler** [comment]
> > > > with this we don't store anywhere the partition mapping right?
> > > 
> > > 
> > > Right. Looking at the code, I think all 3 headers will be available all the time, so we take the first that matches. But I need to verify this in a final booted iso. Up to now I've only tested it by running the challenger binary locally.
> > 
> > this is fine for local iteration but It won't be always the case - only the label is the 'minimum' that is set from the user - all the rest is generated on the node after install
> 
> True. Still the code would still work then. If there is only the label, it will use the label.

This change is fine overall indeed, but we need to also save the state somehow otherwise if we specify the label only when applying  the spec  - install would work, but first boot after install won't

File: config/crd/bases/keyserver.kairos.io_sealedvolumes.yaml (hunk 2 above)
Comment (jimmykarily) [reply config/crd/bases/keyserver.kairos.io_sealedvolumes.yaml]: correct, added now and re-generated the manifests

File: pkg/challenger/challenger_test.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+			})
+
+			It("returns the passphrase", func() {
+				Expect(true).To(BeTrue())
+			})
+		})
+
+		When("a matching sealedvolume doesn't exist", func() {
+			It("returns an error", func() {
+				Expect(true).To(BeTrue())
```
Comment (jimmykarily) [inline pkg/challenger/challenger_test.go:42]: At this point, we need to run the challenger client and make a request to the server for a passphrase. The request is constructed using an actual TPM and I don't see an easy way around that. This means, I need to run the command inside a VM with an emulated TPM (:face_exhaling:). This PR is growing too fast.

Comment (jimmykarily) [reply pkg/challenger/challenger_test.go]: I switched to unit tests for now. They are faster simpler and they do the job. We will introduce an end-to-end suite in another PR

**Review by mudler — APPROVED** [review]
This is looking really good, nice also the test additions :100: :rocket:  !

**Pull request merged**